Sources/src/FromSTMTestBeamData_source.cc

```diff
@@ -135,9 +135,6 @@ namespace mu2e {
   //----------------------------------------------------------------
   void STMTestBeamDataDetail::closeCurrentFile() {
     currentFileName_ = "";
-    currentFile_->close();
```

Comment:
I recommend calling currentFile->reset(); here.  That actually closes the file so that your internal state keeps in sync with the intent of the callback model.

Reply:
Thanks, Rob. Done in 6d6022a85